examples/text-classification/run_pl_glue.py

```diff
@@ -1,115 +1,205 @@
 
-    def __init__(self, hparams):
-        hparams.glue_output_mode = glue_output_modes[hparams.task]
+    def __init__(self, hparams: Namespace):
+        """A Pytorch Lightning Module for training/evaluating transformers models on the GLUE benchmark
+
+        Args:
+            hparams (Namespace): Supplied CLI arguments to configure the run.
+        """
         num_labels = glue_tasks_num_labels[hparams.task]
+        hparams.glue_output_mode = "classification" if num_labels > 1 else "regression"
 
         super().__init__(hparams, num_labels, self.mode)
 
     def forward(self, **inputs):
         return self.model(**inputs)
 
     def training_step(self, batch, batch_idx):
-        inputs = {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[3]}
-
-        if self.config.model_type != "distilbert":
-            inputs["token_type_ids"] = batch[2] if self.config.model_type in ["bert", "xlnet", "albert"] else None
 
-        outputs = self(**inputs)
+        outputs = self(**batch)
         loss = outputs[0]
 
         tensorboard_logs = {"loss": loss, "rate": self.lr_scheduler.get_last_lr()[-1]}
         return {"loss": loss, "log": tensorboard_logs}
 
     def prepare_data(self):
-        "Called to initialize data. Use the call to construct features"
-        args = self.hparams
-        processor = processors[args.task]()
-        self.labels = processor.get_labels()
-
-        for mode in ["train", "dev"]:
-            cached_features_file = self._feature_file(mode)
-            if os.path.exists(cached_features_file) and not args.overwrite_cache:
-                logger.info("Loading features from cached file %s", cached_features_file)
-                features = torch.load(cached_features_file)
-            else:
-                logger.info("Creating features from dataset file at %s", args.data_dir)
-                examples = (
-                    processor.get_dev_examples(args.data_dir)
-                    if mode == "dev"
-                    else processor.get_train_examples(args.data_dir)
+        """
+        Use this to download and prepare data.
+        In distributed (GPU, TPU), this will only be called once.
+        This is called before requesting the dataloaders.
+        """
+        self.dataset = nlp.load_dataset("glue", name=self.hparams.task)
+        self.metric = nlp.load_metric("glue", name=self.hparams.task)
+
+        cached_dataset_file = self._feature_file(self.hparams.task)
+
+        if os.path.exists(cached_dataset_file) and not self.hparams.overwrite_cache:
+            logger.info("Loading from cached dataloader file ", cached_dataset_file)
+            self.dataset = torch.load(cached_dataset_file)
+        else:
+            logger.info("No cache file found. Creating one at", cached_dataset_file)
+
+            # Make the directory to hold cached dataset if it doesn't already exist
+            os.makedirs(os.path.dirname(cached_dataset_file), exist_ok=True)
```

Comment:
```python
from Pathlib import Path
Path(cached_dataset_file).parent.mkdir(exist_ok=True)
```

Reply:
I'm a huge fan of this idea. Kept it as `os` as the other examples all use that, but everyone knows `pathlib` is better. 😄